source/Octopus.Client/Util/ParameterHelper.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Octopus.Client.Exceptions;
+using Octopus.Client.Repositories.Async;
+
+namespace Octopus.Client.Util
+{
+    static class ParameterHelper
+    {
+        public static Dictionary<string, object> CombineParameters(Dictionary<string, object> currentAdditionalQueryParameters, object userProvidedParameters)
+        {
+            if (currentAdditionalQueryParameters == null)
+                currentAdditionalQueryParameters = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+
+            if (userProvidedParameters == null)
+                userProvidedParameters = new { };
+            var resultDictionary = GetParameters(userProvidedParameters);
+
+            // Value from the current repository(currentAdditionalQueryParameters) overrides the one from user (userProvidedParameters)
+            currentAdditionalQueryParameters.ToList().ForEach(x => resultDictionary[x.Key] = x.Value);
+            return new Dictionary<string, object>(resultDictionary);
+        }
+
+        public static void ValidateSpaceParameters(SpaceQueryParameters curreSpaceQueryParameters, bool newIncludeGlobal,
+            string[] newSpaceIds)
+        {
+            if (curreSpaceQueryParameters == null) return;
+
+            if (curreSpaceQueryParameters.IncludeGlobal != newIncludeGlobal)
```

Comment:
This isn't right -> if you are current `includeGlobal = true`, then you should be able to limit this to `includeGlobal = false`